pandas/conftest.py

```diff
@@ -354,7 +354,8 @@ def unique_nulls_fixture(request):
 
 
 TIMEZONES = [None, 'UTC', 'US/Eastern', 'Asia/Tokyo', 'dateutil/US/Pacific',
-             'dateutil/Asia/Singapore']
+             'dateutil/Asia/Singapore', tzutc(), tzlocal(), FixedOffset(300),
+             FixedOffset(0), FixedOffset(-300)]
```

Comment:
Brainstorm:

- non-round numbers for `FixedOffset`
- we have a small number of tests with tzinfos from `psycopg2.tz` and `matplotlib` (not sure where in the matplotlib namespace)
- random subset of `pytz.all_timezones` (592 of them)
- equivalent from `dateutil.tz` (not sure off the top of my head how to access that, but if we figure it out I'll make a PR there)

Reply:
how much additional time do the tests take with these added fixtures? how many more tests

Reply:
- Sure we can add one non-round FixedOffset
- matplotlib UTC is just `datetime.timezones.utc`: https://github.com/matplotlib/matplotlib/blob/9e8be5b0dc0a03f569d7a248f67ef136520adbbb/lib/matplotlib/dates.py#L172
- IIRC we just convert `psycopg2.tz` timezones (from sql functions) directly to UTC, and we don't directly advertise accepting them in other functions. But I guess we do have code that supposedly handles this: https://github.com/pandas-dev/pandas/blob/3702de2545f24e1999cf574bc8dd115460e2db90/pandas/_libs/tslibs/timezones.pyx#L218
- As for more pytz and dateutil timezones, I'd be most interested in testing non-conventional timezones, e.g. odd time shifting, but I feel its not necessary to test all of them (baring testing all of them doesn't take that much additional time)

Reply:
We can add these in a follow up and further discuss.